### PR TITLE
feat(settings): open settings dialog directly

### PR DESCRIPTION
**Pull request by guyi2000**
## Summary

- Problem: The sidebar settings gear opened a three-item menu before users could reach the settings dialog, and settings fields did not consistently show the built-in defaults.
- Why it matters: Settings should be discoverable at a glance, especially for first-time users configuring local runtimes and engine paths.
- What changed: The gear now opens the workspace settings dialog directly on General settings, General/LLM settings display default values, workspace root shows the effective runtime default, and the frontend port field was removed from the visible settings form.
- What did NOT change (scope boundary): Existing backend compatibility for `server.frontendPort` remains; no storage format migration, auth, or execution behavior changed.

## Change Type (select all)

- [ ] Bug fix
- [x] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [ ] Gateway / orchestration
- [ ] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [x] API / contracts
- [x] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #
- Related #
- [ ] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

N/A

## Regression Test Plan (if applicable)

N/A

## User-visible / Behavior Changes

- Clicking the sidebar settings gear now opens the settings dialog directly instead of a three-option menu.
- General settings fields show each field's default value.
- LLM settings show default Base URL / Model information.
- Workspace root now displays the effective default runtime directory.
- The frontend port is no longer shown in the settings form.

## Diagram (if applicable)

```text
Before:
[settings gear] -> [Capabilities / LLM / Database menu] -> [settings panel]

After:
[settings gear] -> [Workspace Settings dialog: General]
```

## Security Impact (required)

- New permissions/capabilities? (`No`)
- Secrets/tokens handling changed? (`No`)
- New/changed network calls? (`No`)
- Command/tool execution surface changed? (`No`)
- Data access scope changed? (`No`)
- If any `Yes`, explain risk + mitigation: N/A

### Environment

- OS: Windows
- Runtime/container: native Node/npm
- Model/provider: N/A
- Integration/channel (if any): frontend settings UI + backend admin settings API
- Relevant config (redacted): local development defaults

### Steps

1. Open the console sidebar and click the settings gear.
2. Open General settings and inspect fields with default hints.
3. Open LLM settings and inspect default Base URL / Model hints.

### Expected

- The settings dialog opens directly without the old three-option menu.
- Defaults are visible next to editable settings.
- Frontend port is not displayed in General settings.

### Actual

- Matches expected in code path and type/build validation.

## Evidence

- [x] Failing test/log before + passing after
- [ ] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

Commands run:

```text
npm run build --prefix backend
npm run type-check --prefix frontend
npm run lint --prefix backend
npm run lint --prefix frontend
npm run test:run --prefix frontend -- tests/stores/slices/preferences.test.ts
```

Notes: frontend lint passed with existing `` warnings in `ai-console.tsx`.

## Human Verification (required)

- Verified scenarios: direct settings gear path in source, no remaining `gearMenu` / `sidebarGearMenu` usage, default value rendering path, backend build, frontend type-check, lint, and preferences store test.
- Edge cases checked: backend still accepts/returns `frontendPort` for compatibility while the frontend no longer renders it.
- What you did **not** verify: no browser screenshot/manual click-through in a running dev server.

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [ ] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? (`Yes`)
- Config/env changes? (`No`)
- Migration needed? (`No`)
- If yes, exact upgrade steps: N/A

## Risks and Mitigations

- Risk: Users who used the gear menu as a shortcut to LLM or Database now need one extra tab click inside the dialog.
  - Mitigation: The settings dialog still exposes those tabs immediately after opening, and the sidebar capability summary shortcut remains available for capabilities.

### CODE REVIEW

File: backend/src/api/admin-settings.ts [hunk 1]
```diff
@@ -123,39 +123,49 @@ type SettingsResponse = {
...
     maxFileSize: 104857600,
-    corsOrigins: '',
-    workspaceRoot: '',
-    checkpointDir: '',
+    corsOrigins: [
+      `http://localhost:${defaultFrontendPort}`,
+      `http://127.0.0.1:${defaultFrontendPort}`,
+      `http://localhost:${defaultBackendPort}`,
+      `http://127.0.0.1:${defaultBackendPort}`,
+    ].join(','),
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/api/admin-settings.ts]: Good catch. This was a real mismatch with `config/index.ts`: the CORS fallback should be derived from the effective configured ports, not only the env/hardcoded port defaults. Updated `buildSettingsResponse()` to compute `effectiveFrontendPort` / `effectiveBackendPort` from `settings.json` overrides first, then fall back to env/defaults, and use those for `defaults.corsOrigins`.

Verified with:

```text
npm run build --prefix backend
npm run type-check --prefix frontend
```

File: backend/src/api/admin-settings.ts [hunk 2]
```diff
@@ -123,39 +123,51 @@ type SettingsResponse = {
...
+    reportsDir: path.join(runtimeBaseDir, 'reports'),
     maxFileSize: 104857600,
-    corsOrigins: '',
-    workspaceRoot: '',
-    checkpointDir: '',
+    corsOrigins: [
+      `http://localhost:${effectiveFrontendPort}`,
+      `http://127.0.0.1:${effectiveFrontendPort}`,
+      `http://localhost:${effectiveBackendPort}`,
+      `http://127.0.0.1:${effectiveBackendPort}`,
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply backend/src/api/admin-settings.ts:157]: Agreed. Even though `getWorkspaceRoot()` already fell back to `runtimeBaseDir`, the exported `config.agentWorkspaceRoot` should represent the same effective default that the settings API displays. Updated `backend/src/config/index.ts` so `agentWorkspaceRoot` falls back to `runtimeBaseDir` directly.

Verified with:

```text
npm run build --prefix backend
npm run type-check --prefix frontend
npm run lint --prefix backend
npm run lint --prefix frontend
npm run test:run --prefix frontend -- tests/stores/slices/preferences.test.ts
```

File: frontend/src/components/settings/llm-settings-panel.tsx [hunk 3]
```diff
@@ -242,6 +250,9 @@ export function LlmSettingsPanel() {
                   required
                 />
               </div>
+              <p className="mt-1 break-all text-xs leading-5 text-muted-foreground">
+                {t('generalSettingsDefaultValuePrefix')}: <span className="font-mono">{defaultBaseUrl}</span>
+              </p>
```
[the remark this answers is not in the document]
Comment (guyi2000) [reply frontend/src/components/settings/llm-settings-panel.tsx]: Good point. Moved the default-value hint rendering into a shared `DefaultValueHint` component under `frontend/src/components/settings/default-value-hint.tsx` and reused it from both General and LLM settings, so the formatting and wrapping behavior stay consistent.

Verified with:

```text
npm run build --prefix backend
npm run type-check --prefix frontend
npm run lint --prefix backend
npm run lint --prefix frontend
npm run test:run --prefix frontend -- tests/stores/slices/preferences.test.ts
```

**Review by qinsz01 — APPROVED** [review]
